kat/src/ecc384_kat.rs

```diff
@@ -12,115 +12,94 @@ Abstract:
 
 --*/
 
+use crate::caliptra_err_def;
 use caliptra_lib::{
-    caliptra_err_def, Array4x12, Array4xN, CaliptraResult, Ecc384, Ecc384Data, Ecc384PrivKeyIn,
-    Ecc384PubKey, Ecc384Scalar, Ecc384Signature,
+    Array4xN, CaliptraResult, Ecc384, Ecc384Data, Ecc384PrivKeyIn, Ecc384PubKey, Ecc384Signature,
 };
 
 caliptra_err_def! {
-    Ecc384,
+    Ecc384Kat,
     Ecc384KatErr
     {
         SignatureGenerateFailure = 0x01,
         SignatureVerifyFailure = 0x02,
+        SignatureMismatch = 0x03,
     }
 }
 
+const SIGNATURE: Ecc384Signature = Ecc384Signature {
+    r: Array4xN([
+        0x36f85014, 0x6f400443, 0x848cae03, 0x57591032, 0xe6a395de, 0x66e7261a, 0x38049fb,
+        0xee15db19, 0x5dbd9786, 0x9439292a, 0x4f5792e4, 0x3a1231b7,
+    ]),
+    s: Array4xN([
+        0xeeea4294, 0x82fd8fa9, 0xd4d5f960, 0xa09edfa6, 0xc765efe5, 0xff4c17a5, 0x12e694fa,
+        0xcc45d3f6, 0xfc3d3b5c, 0x62739c1f, 0xb9fcae3, 0x26f54b43,
+    ]),
+};
+
 #[derive(Default, Debug)]
 pub struct Ecc384Kat {}
 
-const SIGNATURE_R: [u32; 12] = [
-    0x36f85014, 0x6f400443, 0x848cae03, 0x57591032, 0xe6a395de, 0x66e7261a, 0x38049fb, 0xee15db19,
-    0x5dbd9786, 0x9439292a, 0x4f5792e4, 0x3a1231b7,
-];
-const SIGNATURE_S: [u32; 12] = [
-    0xeeea4294, 0x82fd8fa9, 0xd4d5f960, 0xa09edfa6, 0xc765efe5, 0xff4c17a5, 0x12e694fa, 0xcc45d3f6,
-    0xfc3d3b5c, 0x62739c1f, 0xb9fcae3, 0x26f54b43,
-];
-
 impl Ecc384Kat {
-    // This function executes the Known Answer Tests (aka KAT) for ECC384.
-    //
-    // Test vector source:
-    // Generated using MbedTLS library.
-    //
-    // # Arguments
-    //
-    /// * None
+    /// This function executes the Known Answer Tests (aka KAT) for ECC384.
     ///
-    /// # Returns
+    /// Test vector source:
+    /// Generated using MbedTLS library.
     ///
-    /// * `CaliptraResult` - Result denoting the KAT outcome.
-    pub fn execute(&self) -> CaliptraResult<()> {
-        self.kat_signature_generate()?;
-        self.kat_signature_verify()?;
-        Ok(())
-    }
-
-    // Performs the signature generation operation.
-    //
-    // # Arguments
-    //
-    /// * None
+    /// # Arguments
+    ///
+    /// * `ecc` - ECC-384 Driver
     ///
     /// # Returns
     ///
     /// * `CaliptraResult` - Result denoting the KAT outcome.
-    fn kat_signature_generate(&self) -> CaliptraResult<()> {
+    pub fn execute(&self, ecc: &Ecc384) -> CaliptraResult<()> {
+        self.kat_signature_generate(ecc)?;
+        self.kat_signature_verify(ecc)
+    }
+
+    fn kat_signature_generate(&self, ecc: &Ecc384) -> CaliptraResult<()> {
         let priv_key = Array4xN([
```

Comment:
Can we make all Array4xN constants. specifically the one within methods that have expected value. This is taking up lot of code space. Consts get promoted readonly sections where as for let bindings the compiler is generating store instructions to the stack.